pkgs/development/libraries/dbus/default.nix

```diff
@@ -43,10 +46,22 @@ stdenv.mkDerivation rec {
       --replace 'DBUS_DAEMONDIR"/dbus-daemon"' '"/run/current-system/sw/bin/dbus-daemon"'
   '';
 
-  outputs = [ "out" "dev" "lib" "doc" ];
+  preConfigure = ''
+    for file in doc/*.xml doc/*.xml.in; do
+      substituteInPlace "$file" --replace "http://www.oasis-open.org/docbook/xml/4.4/docbookx.dtd" \
+        "${docbook_xml_dtd_44}/xml/dtd/docbook/docbookx.dtd"
+    done
+  '';
+
+  preBuild = ''
+    makeFlagsArray+=XMLTO="${xmlto}/bin/xmlto -x ${docbook_xsl}/xml/xsl/docbook/manpages/docbook.xsl"
```

Comment:
Are these necessary? Would not just adding `docbook_xml_dtd_44` and `docbook-xsl-nons` to `nativeBuildInputs` work? That should add them from `$XMLCATALOG`.

Reply:
Done, thanks! I wasn't sure how to deal with this and that is much cleaner.